infra/feast-operator/internal/controller/services/util.go

```diff
@@ -0,0 +1,79 @@
+package services
+
+import (
+	"github.com/feast-dev/feast/infra/feast-operator/api/feastversion"
+	feastdevv1alpha1 "github.com/feast-dev/feast/infra/feast-operator/api/v1alpha1"
+)
+
+func IsLocalRegistry(featureStore *feastdevv1alpha1.FeatureStore) bool {
+	appliedServices := featureStore.Status.Applied.Services
+	return appliedServices != nil && appliedServices.Registry != nil && appliedServices.Registry.Local != nil
+}
+
+func ApplyDefaultsToStatus(cr *feastdevv1alpha1.FeatureStore) error {
+	cr.Status.FeastVersion = feastversion.FeastVersion
+	applied := cr.Spec.DeepCopy()
+	if applied.Services == nil {
+		applied.Services = &feastdevv1alpha1.FeatureStoreServices{}
+	}
+
+	// default to registry service deployment
+	if applied.Services.Registry == nil {
+		applied.Services.Registry = &feastdevv1alpha1.Registry{}
+	}
```

Comment:
nit ... since we're doing so much here with services, maybe this section (and what follows) cleans up a little if we -
```suggestion
        services := applied.Services
	if services == nil {
		services = &feastdevv1alpha1.FeatureStoreServices{}
	}

	// default to registry service deployment
	if services.Registry == nil {
		services.Registry = &feastdevv1alpha1.Registry{}
	}
```
etc.

Reply:
i realize this is leftover from one of my prior PRs, but now that we're doing so much more with that pointer i think it makes sense